app/src/main/java/ar/com/wolox/android/example/ui/news/NewsViewHolderAdapter.kt

```diff
@@ -31,27 +41,31 @@ class NewsViewHolderAdapter : ListAdapter<News, NewsViewHolderAdapter.ViewHolder
         this.newsList.addAll(newsList)
     }
 
-    fun clear() {
-        this.newsList.clear()
-    }
-
-    fun addNews(newsList: List<News>) {
-        this.newsList.addAll(newsList)
-    }
-
     override fun getItemCount(): Int {
         return newsList.size
     }
 
     override fun onBindViewHolder(holder: ViewHolder, position: Int) {
         val news = newsList[position]
+        val userId = sharedPref.getInt(context.resources.getString(R.string.login_user_id), 0)
+
+        news.like = news.likes?.contains(userId) == true
 
         holder.apply {
             vNewsTitle.text = news.title
             vNewsBody.text = news.text
             vTimeAgo.text = news.getTimeFormated()
-
             vNewsImage.setImageURI(news.picture)
+            vFavButton.isChecked = news.like
+
+            vFavButton.setOnClickListener {
+                listener.onToggleClick(news)
+                news.like = !news.like
+            }
+
+            vNewsContainer.setOnClickListener {
+                NewsDetailActivity.startNewsDetailActivity(context, news)
```

Comment:
Well, this is not OK. If an element is clicked inside an adapter, you have to call a Listener and inform Fragment. Fragment will inform Presenter and finally you're going to new activity.